### PR TITLE
include type promotion info in torch.add/mul/div documentation

### PR DESCRIPTION
**Pull request by nairbv**
Related to https://github.com/pytorch/pytorch/pull/25114 but would like to improve torch.add doc on release branch.

Issue: https://github.com/pytorch/pytorch/issues/26825

FYI: @xuhdev

### CODE REVIEW

File: torch/_torch_docs.py [hunk 1]
```diff
@@ -143,57 +143,44 @@ def merge_dicts(*dicts):
 
 add_docstr(torch.add,
            r"""
-.. function:: add(input, other, out=None)
+.. function:: add(input, other, alpha=1, out=None)
```
Comment (gchanan) [inline torch/_torch_docs.py]: Why did you combine these functions?  I don't think it's good in the documentation to start with "here's the most advanced form that you will never use".

Comment (nairbv) [reply torch/_torch_docs.py]: I had removed it to be consistent with the related PR which also removed it, but now re-included both variants in this PR.

**Comment by zou3519** [comment]
I'm confused -- does this supercede https://github.com/pytorch/pytorch/pull/25114?

File: torch/_torch_docs.py [hunk 2]
```diff
@@ -145,21 +145,21 @@ def merge_dicts(*dicts):
            r"""
 .. function:: add(input, other, out=None)
 
-Adds the scalar :attr:`other` to each element of the input :attr:`input`
-and returns a new resulting tensor.
+Each element of the tensor ``other`` is added to each element of the
```
Comment (zou3519) [inline torch/_torch_docs.py]: To be consistent with the rest of our documentation, ese should be `:attr:`other`` and `:attr:`input``

File: torch/_torch_docs.py [hunk 3]
```diff
@@ -145,21 +145,21 @@ def merge_dicts(*dicts):
...
-
-Args:
-    {input}
-    value (Number): the number to be added to each element of :attr:`input`
+The shapes of :attr:`input` and :attr:`other` must be
+:ref:`broadcastable <broadcasting-semantics>`.
 
-Keyword arguments:
-    {out}
+The :class:`torch.dtype` of the result tensor is the result of calling
```
Comment (zou3519) [inline torch/_torch_docs.py]: I want to point out that numpy makes no mention of type promotion in their doc: https://docs.scipy.org/doc/numpy/reference/generated/numpy.add.html

File: torch/_torch_docs.py [hunk 4]
```diff
@@ -145,21 +145,21 @@ def merge_dicts(*dicts):
...
+:meth:`torch.result_type` with the :attr:`input` and :attr:`other`
```
Comment (zou3519) [inline torch/_torch_docs.py]: As a user, I don't want to click through and understand `torch.result_type` before using `torch.add`. Maybe say something like "if the input and other have different dtype, the output has a different dtype according to type promotion rules"

File: torch/_torch_docs.py [hunk 5]
```diff
@@ -169,27 +169,25 @@ def merge_dicts(*dicts):
...
 
 .. math::
     \text{{out}} = \text{{input}} + \text{{alpha}} \times \text{{other}}
 
-If :attr:`other` is of type FloatTensor or DoubleTensor, :attr:`alpha` must be
-a real number, otherwise it should be an integer.
+If the result of the addition between :attr:`input` and :attr:`other` is:
+- floating point, :attr:`alpha` must be `float` or `int`
+- integral, :attr:`alpha` must be `int`
+- boolean, :attr:`alpha` must be `int` or `bool.
```
Comment (zou3519) [inline torch/_torch_docs.py]: missing backtick for bool

**Comment by xuhdev** [comment]
> I'm confused -- does this supercede #25114?

This PR is for the release branch. #25114 is for master

File: torch/_torch_docs.py (hunk 5 above)
Comment (nairbv) [reply torch/_torch_docs.py]: fixed

File: torch/_torch_docs.py (hunk 4 above)
Comment (nairbv) [reply torch/_torch_docs.py]: reworded

File: torch/_torch_docs.py [hunk 6]
```diff
@@ -145,21 +145,22 @@ def merge_dicts(*dicts):
...
 
-If :attr:`input` is of type FloatTensor or DoubleTensor, :attr:`other` must be
-a real number, otherwise it should be an integer.
+The shapes of :attr:`input` and :attr:`other` must be
+:ref:`broadcastable <broadcasting-semantics>`.
 
-Args:
-    {input}
-    value (Number): the number to be added to each element of :attr:`input`
+If the :class:`torch.dtype` of :attr:`input` and :attr:`other` differ,
```
Comment (gchanan) [inline torch/_torch_docs.py]: does this "if" add anything?  The the rules of type promotion apply even if the dtypes are the same, right?

File: torch/_torch_docs.py [hunk 7]
```diff
@@ -1644,12 +1643,19 @@ def merge_dicts(*dicts):
 .. math::
     \text{out}_i = \frac{\text{input}_i}{\text{other}}
 
-If :attr:`input` is of type `FloatTensor` or `DoubleTensor`, :attr:`other`
-should be a real number, otherwise it should be an integer
+If the :class:`torch.dtype` of :attr:`input` and :attr:`other` differ,
```
Comment (gchanan) [inline torch/_torch_docs.py]: same here.

File: torch/_torch_docs.py [hunk 8]
```diff
@@ -3398,8 +3404,14 @@ def merge_dicts(*dicts):
 .. math::
     \text{out}_i = \text{other} \times \text{input}_i
 """ + r"""
-If :attr:`input` is of type `FloatTensor` or `DoubleTensor`, :attr:`other`
-should be a real number, otherwise it should be an integer
+
+If the :class:`torch.dtype` of :attr:`input` and :attr:`other` differ,
```
Comment (gchanan) [inline torch/_torch_docs.py]: same here.

**Review by gchanan — COMMENTED** [review]
sorry, getting to this late.

**Pull request merged**